src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/admin/SessionTemplateExcludeDatesController.kt

```diff
@@ -0,0 +1,143 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.controller.admin
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.tags.Tag
+import jakarta.validation.Valid
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.visitscheduler.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.ExcludeDateDto
+import uk.gov.justice.digital.hmpps.visitscheduler.service.SessionTemplateService
+import java.time.LocalDate
+
+const val SESSION_TEMPLATE_EXCLUDE_DATE_PATH: String = "$SESSION_TEMPLATE_PATH/{reference}/exclude-date"
+
+const val ADD_SESSION_TEMPLATE_EXCLUDE_DATE: String = "$SESSION_TEMPLATE_EXCLUDE_DATE_PATH/add"
+
+const val REMOVE_SESSION_TEMPLATE_EXCLUDE_DATE: String = "$SESSION_TEMPLATE_EXCLUDE_DATE_PATH/remove"
+const val GET_SESSION_TEMPLATE_EXCLUDE_DATES: String = SESSION_TEMPLATE_EXCLUDE_DATE_PATH
+
+@RestController
+@Validated
+@Tag(name = "Session Template exclude dates rest controller")
+@RequestMapping(name = "Session Template Exclude Dates Configuration Resource", produces = [MediaType.APPLICATION_JSON_VALUE])
+class SessionTemplateExcludeDatesController(
+  private val sessionTemplateService: SessionTemplateService,
+) {
+  @PreAuthorize("hasAnyRole('VISIT_SCHEDULER_CONFIG', 'VISIT_SCHEDULER')")
```

Comment:
2 roles currently, once we move this to staff service can remove VISIT_SCHEDULER_CONFIG authorization.